server/src/main/java/org/elasticsearch/search/rank/rrf/RRFRankShardContext.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank.rrf;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.search.query.QueryPhase;
+import org.elasticsearch.search.query.QueryPhaseExecutionException;
+import org.elasticsearch.search.rank.RankShardContext;
+import org.elasticsearch.search.rank.RankShardResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RRFRankShardContext extends RankShardContext {
+
+    protected final int windowSize;
+
+    public RRFRankShardContext(List<Query> queries, int size, int from, int windowSize) {
+        super(queries, size, from);
+        this.windowSize = windowSize;
+    }
+
+    @Override
+    public void executeQueries(SearchContext searchContext) {
+        try {
+            RRFRankSearchContext rrfRankSearchContext = new RRFRankSearchContext(searchContext);
+            QueryPhase.executeInternal(rrfRankSearchContext);
+
+            List<TopDocs> rrfRankResults = new ArrayList<>();
+            rrfRankSearchContext.windowSize(windowSize);
+            for (Query query : queries) {
+                rrfRankSearchContext.rrfRankQuery(query);
+                QueryPhase.executeInternal(rrfRankSearchContext);
+                rrfRankResults.add(rrfRankSearchContext.queryResult().topDocs().topDocs);
```

Comment:
We should do a merge here with rank position in the sets. Right now this returns `windowSize * queries.size()` number of docs. It really only needs to be topDocs `windowSize` with their rank metadata

Reply:
I can do that. I assumed for correctness it was better to send all the results back and only do a single ranking. If this is the case we can probably just only have a single top docs and may not need changes to `QueryShardResult` at all.

Reply:
I split this up and redid the ranking. `window_size` now applies to both the shard and the coordinator.